### ISSUE TITLE
Example code

### ISSUE THREAD

**Issue by zhangy76** [robot-acceleration/GRiD#1]
Hi,

I wonder if there is any example code for using the GRiD?

Thanks

**Comment by plancherb1** [comment]
You can find some examples in timeGRiD.cu of using the CUDA code and in timeGRiD.py of invoking GRiD to generate the CUDA code here: https://github.com/robot-acceleration/GRiDBenchmarks

If you have thoughts on what examples would be the most useful — I intend to do some work on GRiD this summer to add some more features etc. so happy to try to add some examples or documentation for examples as well!

**Comment by zhangy76** [comment]
Thank you for the reply! Releasing of additional examples and documents would definitly help! 

In fact,t I find the repository online as I am trying to incorporate the rigid body dynamics algorithm into pytorch. May I quickly ask you if you think this would be possible if using GRiD?

Thanks

**Comment by plancherb1** [comment]
I think it totally should be -- I haven't used pytorch myself but I see no reason why that shouldn't work. If you find that you need different functions exposed let me know and I can work to try to get that going. I'd love to get more use cases and examples up this summer so very happy to send some time trying to expose APIs for what you need.

That said, if your primary use case is training large NNs and you want to run thousands of simultaneous simulations you may find that [Brax](https://github.com/google/brax) may be what you are looking for as Brax is designed and optimized for thousands of simultaneous simulations while GRiD is designed and optimized for tens to hundreds. Also here is a link to a [colab]((https://colab.research.google.com/github/google/brax/blob/main/notebooks/training_torch.ipynb) using it with pytorch.

**Comment by zhangy76** [comment]
Hi,

Thanks for the reply and pointing out Brax!

I am interested in training the deep learning models. Brax is a good choice for large scale training but Brax has several limitations (my understanding may be wrong):
- URDF file can not be easily imported
- The imported models can not vary within a batch, e.g. the input data within a minibatch have different weights or different scales. I am wondering if we can easily scale body size and set weights?
- Following the previous point, Brax is not able to backpropagate the gradients to mass and body scale (a physics enginee named NimblePhysics can realize these functions but it can not do batch processing).

**Comment by plancherb1** [comment]
Yah part of the reason I started working on my own library is that the existing ones didn't do what I wanted -- another new library that may be of use to you is https://github.com/erwincoumans/tiny-differentiable-simulator -- it definitely has the gradients but/and not sure if it has the other features!

**Comment by zhangy76** [comment]
I will take a look at that and thanks! I will close the issue for now. Looking forward to seeing the updates of this repository.
